world/petitions/models.py

```diff
@@ -264,6 +301,15 @@ def mark_posts_read(self, dompc):
             participant.save()
         except PetitionParticipation.DoesNotExist:
             pass
+    def mark_posts_unread(self, dompc):
+        """If dompc is a participant, mark their posts read"""
+        try:
+            participants = self.petitionparticipation_set.all().exclude(dompc=dompc)
+            for participant in participants:
+                    participant.unread_posts = True
+                    participant.save()
+        except PetitionParticipation.DoesNotExist:
```

Comment:
DoesNotExist is never raised by a filter/exclude statement, so no need for the exception block here